src/protocol.rs

```diff
@@ -407,6 +416,127 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    #[cfg(test)]
```

Comment:
nit: this isn't necessary, you can remove it.

The whole test module is already tagged with `#[cfg(test)]`, so we don't have to flag individual tests as well. You can read [the Rust book chapter on tests](https://doc.rust-lang.org/book/ch11-00-testing.html) and especially the [section on test organization](https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-module-and-cfgtest) for details on the standard organization and labelling of tests.